CardScan/Classes/CreditCardOcr/MainLoopStateMachine.swift

```diff
@@ -0,0 +1,59 @@
+//
+//  MainLoopStateMachine.swift
+//  CardScan
+//
+//  Created by Sam King on 8/5/20.
+//
+
+import Foundation
+
+public enum MainLoopState {
+    case initial
+    case ocrOnly
```

Comment:
have you considered adding state to these? In the android implementation, I maintained all state, including the voting algorithm, in these states. So when transitioning to an `ocrOnly` state, the state would receive an initial PAN value. When events come in, the state itself keeps track of the current vote.

Reply:
See the CardScan state machine https://github.com/getbouncer/cardscan-android/blob/master/cardscan-ui/src/main/java/com/getbouncer/cardscan/ui/result/MainLoopAggregator.kt#L79

and the liveness state machine https://github.com/getbouncer/cardverify-android/blob/master/liveness-ui/src/main/java/com/getbouncer/liveness/ui/result/MainLoopAggregator.kt#L195

Reply:
This allows the state to be the single `truth` of the system. there's no need to correlate these state enums with the values stored for voting.